opennms-doc/guide-admin/src/asciidoc/text/telemetryd/introduction.adoc

```diff
@@ -2,68 +2,80 @@
 
 === _Listeners_ and _Parsers_
-_Listeners_ are responsible for receiving sensor data from some external source and hand them off to _Parsers_ which handles the transport protocol format and extracts the transported data.
+_Listeners_ receive sensor data from an external source and hand them off to _Parsers_, which handle the transport protocol format and extract the transported data.
 
 A _Listener_ does the transport protocol handling like opening a socket and accepting incoming connections.
-The received messages are then passed to the parser configured for the _Listener_ which parses the concrete protocol implementation and extract all available information out of the received message.
+The received messages are then passed to the parser configured for the _Listener_ which parses the concrete protocol implementation and extracts all available information out of the received message.
 For further processing, the extracted data is handed over to the configured queue.
 
-For example: A _Listener_ may open a UDP socket for receiving packets and a assigned _Parser_ parses the incoming IPFix packets.
+For example: a _Listener_ may open a UDP socket for receiving packets and an assigned _Parser_ parses the incoming IPFIX packets.
 
-_Parsers_ and _Listeners_ must work together in a tight manner.
-Therefore they can not be combined in any possible combination.
-Which _Parser_ is supported by which _Listener_ is described in the documentation of each _Parser_ accordingly.
+_Parsers_ and _Listeners_ must work together closely.
+Therefore, they can not be combined in any random way.
+Which _Parser_ is supported by which _Listener_ is described in the documentation of each respective _Parser_.
 
-TIP: You can provide the parameter `maxClockSkew` in your parser definition to enable clock skew detection for exporters. It specifies the maximum delta in seconds between exporter and Minion timestamps. If violated an alarm will be generated for the exporting device. The default value is 0, so clock skew detection is disabled. Furthermore, a parameter `clockSkewEventRate` can be used to rate-limit clock skew events. The default is `3600` seconds, so every hour an event will be sent.
+TIP: You can provide the parameter `maxClockSkew` in your parser definition to enable clock skew detection for exporters. 
+It specifies the maximum delta in seconds between exporter and Minion time stamps. 
+If violated, an alarm will be generated for the exporting device. 
+The default value is 0, so clock skew detection is disabled. 
+Furthermore, a parameter `clockSkewEventRate` can be used to rate-limit clock skew events.
+The default is `3600` seconds, so every hour an event will be sent.
+
+==== Working with Minions
+While {opennms-product-name} can operate as both a listener and a parser, Minion cannot.
+A Minion receives data (listens). 
+Listeners on Minion must use parsers on {opennms-product-name} or Sentinel to process the data. 
```

Comment:
Is this true? As far I remember we have parsers on the Minion as described here: https://docs.opennms.org/opennms/releases/26.1.1/guide-admin/guide-admin.html#_push_sensor_data_through_minion

@fooker can you verify this here?

Reply:
@indigo423 has a point there. The listeners and parsers run on the minion whereas the adapters are executed on the sentinel. The main opennms instance can be both.

Reply:
Thanks for clarifying. Am updating the documentation.